src/renderer/ssf-api.ts

```diff
@@ -236,6 +237,28 @@ export class SSFApi {
         }
     }
 
+    /**
+     * Allows JS to register a log retriever that can be used by electron main process.
+     * @param  {Object} retriever  function that can be called to retrieve logs
+     */
+    public registerLogRetriever(retriever): void {
```

Comment:
What is the contract for the retriever function? Does it have to call `logRecevier`? How many times?

If so, I think it would be better to have the function return the logs, maybe as a promise to allow for asynchronous collection of logs. The interface will be much easier to understand, and it's harder to forget to return logs by mistake.

Reply:
When the log retriever is called, the logs are sent all in one go as a message. Not sure I follow how you want it to be done instead - do you know of any place in the code where it's done like that which I could look at for reference?

Reply:
Instead of registering one callback of type `void retreiveLogs()` that is required to call another global method, `sendLogs()`, have the callback function return the logs.

That reduces the number of global methods in the API, and makes the callback easier to get right, since it is clear from the signature that it should return the logs.

It also avoids the problem of how to handle calls to `sendLogs()` when no one is collecting any logs. Because there won't be any method.
It will also help to correlate the number of calls to registered log sources, and their responses.

Reply:
Ok, that make sense, but I don't know what that would look like.